adapters/adhese/adhese.go

```diff
@@ -0,0 +1,282 @@
+				ID:     adheseBid.Id,
+				ImpID:  adheseBid.CreativeName,
+				DealID: adheseBid.OrderId,
+				Price:  price,
+				W:      width,
+				H:      height,
+				CID:    adheseBid.OrderId,
+				CrID:   adheseBid.Id,
+				NURL:   adheseBid.ImpressionCounter,
+				BURL:   adheseBid.Tracker,
+				AdM:    getAdMarkup(adheseBid),
+				Ext:    adheseObj,
+			}},
+			Seat: "",
+		}},
+		BidID: adheseBid.OrderId,
+		Cur:   adheseBid.Extension.Prebid.Cpm.Currency,
+	}
+}
+
+func convertAdheseOpenRtbBid(adheseBid openrtb_ext.AdheseOpenRtbBid) openrtb.BidResponse {
+	price, _ := strconv.ParseFloat(adheseBid.Extension.Prebid.Cpm.Amount, 64)
+	width, _ := strconv.ParseUint(adheseBid.Width, 10, 64)
+	height, _ := strconv.ParseUint(adheseBid.Height, 10, 64)
+	var response openrtb.BidResponse = adheseBid.OriginData
+	response.ID = adheseBid.Origin
+	if adheseBid.OriginInstance != "" {
+		response.ID = response.ID + "-" + adheseBid.OriginInstance
+	}
+	if len(response.SeatBid) > 0 && len(response.SeatBid[0].Bid) > 0 {
+		response.SeatBid[0].Bid[0].Price = price
+		response.SeatBid[0].Bid[0].W = width
+		response.SeatBid[0].Bid[0].H = height
+		response.SeatBid[0].Bid[0].AdM = adheseBid.Body
+		if ContainsAny(adheseBid.Body, []string{"<script", "<div", "<html"}) {
+			response.SeatBid[0].Bid[0].AdM += "<img src='" + adheseBid.ImpressionCounter + "' style='height:1px; width:1px; margin: -1px -1px; display:none;'/>"
+		}
+	}
+
+	response.Cur = adheseBid.Extension.Prebid.Cpm.Currency
+	return response
+}
+
+func getAdMarkup(adheseBid openrtb_ext.AdheseBid) string {
+	if adheseBid.Ext == "js" && ContainsAny(adheseBid.Body, []string{"<script", "<div", "<html"}) {
+		return adheseBid.Body + "<img src='" + adheseBid.ImpressionCounter + "' style='height:1px; width:1px; margin: -1px -1px; display:none;'/>"
+	} else if adheseBid.Ext == "js" && ContainsAny(adheseBid.Body, []string{"<?xml", "<vast"}) {
+		return adheseBid.Body
+	} else {
+		return adheseBid.Tag
+	}
+}
+
+func getBidType(bidAdm string) openrtb_ext.BidType {
+	if bidAdm != "" && ContainsAny(bidAdm, []string{"<?xml", "<vast"}) {
+		return openrtb_ext.BidTypeVideo
+	}
+	return openrtb_ext.BidTypeBanner
+}
+
+func WrapError(errorStr string) *errortypes.BadInput {
+	return &errortypes.BadInput{Message: errorStr}
+}
+
+func ContainsAny(raw string, keys []string) bool {
+	for i := 0; i < len(keys); i++ {
+		if strings.Contains(strings.ToLower(raw), keys[i]) {
+			return true
+		}
+	}
+	return false
+
+}
+
+func NewAdheseAdapter(config *adapters.HTTPAdapterConfig, uri string) *AdheseAdapter {
+	return NewAdheseBidder(adapters.NewHTTPAdapter(config).Client, uri, 0)
+}
+
+// Set dummyCacheBuster to 0 in order to generate a cache buster
+func NewAdheseBidder(client *http.Client, uri string, dummyCacheBuster int) *AdheseAdapter {
+	template, _ := template.New("endpointTemplate").Parse(uri)
+	return &AdheseAdapter{http: &adapters.HTTPAdapter{Client: client}, endpointTemplate: *template, dummyCacheBuster: dummyCacheBuster}
```

Comment:
dummyCacheBuster not in use, right? If so to be removed from all over the place.